CartoDB/cartodb
# Changes for adding automatic classes for any analysis form

Related ticket: #12095.

Basically, we need to add track classes in all analysis form editor components. With that, we could check if they are working as expected at UI perspective.

## Pending things
- [x] Decide the good way to add them in the current analysis form "workflow".
- [x] Check all backbone form editors will have this class added when needed (check `_setOptions` method is called in their respective initialize'rs').
- [x] Check the behavior for the nested forms.
- [x] Add necessary tests.

## Steps to test it
We have changed the way we set the options in any editor component, so I encourage you to test all components in different contexts. One, withing an analysis form (with track class), and another out of the analysis world. I've created a table with all Backbone.Form.editors available in Builder, providing a place to check in an analysis and another out of it.

| Reviewed | Name | Analysis to find it (with track class)  | Place to find it (without track class)  |
|---|---|---|---|
| [x] | CodeEditor  |  Geocoding by street address (advanced mode)  | _x_  | 
| [-] | DataObservatoryDropdown  | Data-observatory :D  | _x_  | 
| [x] | DateTime  | _x_  | Editing a feature  | 
| [x] | Enabler  | Connect with lines  | _x_  | 
| [x] | EnablerEditor  | Centroid  | Editing a widget, in the form  | 
| [x] | Fill  | _x_  | Style a layer  | 
| [-] | List  | _x_  | _x_  |
| [x] | MultiSelect  | Markov  | _x_  |
| [x] | NodeDataset  | _Any analysis_  | _x_  |
| [x] | Number  | Area of influence  | Export image form  |
| :boom: | Operators  | Intersection  | Aggregation options when style is aggregated |
| [x] | Radio  | Merge  | Widget form view |
| [x] | Select  | Closest  | Style form view, selecting a label  |
| [-] | Slider  | _x_  | _x_  |
| [x] | SortableList  | _x_  | Customizing a legend  |
| [x] | Suggest  | _x_  | Editing a feature |
| [x] | Taglist  | _x_  | Map metadata form view (within the modal) |
| [x] | Text  | Filter analysis | Widget edit form view |
| [x] | TextArea  | _x_ | Formula widget edit form view |
| [x] | Toggle  | Georeference by street address (advance mode is the toggle) | _x_ |

**Notes**
There are several Backbone.Forms editors not migrated yet: `[Checkbox, Checkboxes, Date, Hidden, Password]`. And also there are some those don't need anything related with classes `[NestedModel, Base, Object,...]`.

**Legend**
x -> There is no place using that editor.

- [x] Apart from that, we have to check that the apply button in any analysis form view has the proper `track` class.
